src/xdocs/contributing.xml

```diff
@@ -127,26 +127,27 @@
       </p>
 
       <p>
-        Please use <a href="https://help.github.com/articles/using-pull-requests/">Pull Request</a>
-        feature of <a href="https://github.com/">Github</a>.<br/>
-        Please provide wide description of update with detailed explanation of problem
-        and how you propose to resolve it.
-        <b>ATTENTION:</b> Please recheck that in your Pull Request there is ONLY your changes and
-        all of them are rebased from most recent HEAD of our master branch.
+        Please create a <a href="https://help.github.com/articles/using-pull-requests/">Pull Request</a>
+        for your contribution. In the pull request description add explanation of the implementation
+        nuances. Please read Pull Request template for more requirements.
+        <b>ATTENTION:</b> Please verify that the Pull Request contains ONLY your intended changes and
+        is based on the most recent HEAD of our master branch.
       </p>
 
       <p>
-        <b>ATTENTION:</b> Please provide single-line meaningful message for commit with reference
-        to github issue or Pull Request for more details.<br/>
-        Ideal format for message is "Issue #Number: Brief single-line message ..... .". Where
-        NUMBER is Issue or Pull Request number at github.
-        <a href="https://github.com/checkstyle/checkstyle/commit/9303fd9d971eb55cdfd62686ba2f5698edb2c83e">Good example</a>
+        <b>ATTENTION:</b>The commit message must adhere to a certain format.
+         It should be "Issue #Number: Brief single-line message", where NUMBER is the issue number at Github
+         (see <a href="https://github.com/checkstyle/checkstyle/commit/9303fd9d971eb55cdfd62686ba2f5698edb2c83e">an example</a>).
+         Small changes of configuration files, documentation fixes etc. can be contributed by starting
+         the commit message with one of "minor:", "config:", "infra:", "doc:" or "spelling:",
+         followed by a blank and a brief single-line message.
       </p>
 
       <p>
-        After you do Pull Request please recheck that where are no problems from our build systems:
-        <a href="https://travis-ci.org/checkstyle/checkstyle">Travis</a> and others ... .
-        You will see that your Pull Request on Github is marked by that jobs automatically in few minutes.
+        After submitting a pull request, it will be automatically checked by
+        <a href="https://travis-ci.org/checkstyle/checkstyle">Travis</a> and some other continuous integration tools.
+        Therefore please recheck after some minutes that the system didn't find any issues.
+        If there are issues, please fix them and provide an updated version of the same pull request.
```

Comment:
> please fix them

This should be expanded on by saying to amend the previous commit. Lots of people give us fixes just by adding a second commit.

Reply:
a bit extended.